### PR TITLE
Improve the navigation endpoint by being able to return the items nav…

### PR DESCRIPTION
**Pull request by sneridagh**
…igation tree providing a navigation.depth parameter

### CODE REVIEW

**Comment by sneridagh** [comment]
@fulv I borrowed and adapted some code from collective.restapi.navigationtree. I hope you don't mind, I also mentioned you on the Changelog. Please let me know if you might want to add/change something.

File: CHANGES.rst [hunk 1]
```diff
@@ -4,7 +4,10 @@ Changelog
 1.5.1 (unreleased)
 ------------------
 
-- Nothing changed yet.
+- Improve the navigation endpoint by being able to return the items navigation
+  tree providing a `navigation.depth` parameter. Borrowed and adapted code from
```
Comment (tisto) [inline CHANGES.rst]: @sneridagh why do we use navigation.depth and not just "depth"? Did you try if that works with expansion as well?

**Comment by fulv** [comment]
@sneridagh great!  My next step was literally to do what you just did, and replace the whole dependency on webcouturier.dropdownmenu with a simple depth request parameter.

**Comment by sneridagh** [comment]
@tisto @gforcada We have a problem with the build and code-analysis:

```
Installed /tmp/easy_install-orgoSu/flake8-print-3.1.0/.eggs/pytest_runner-4.2-py2.7.egg
Got flake8-print 3.1.0.
Getting distribution for 'pycodestyle'.
warning: no previously-included files matching '*.pyc' found under directory 'docs'
warning: no previously-included files matching '*.pyo' found under directory 'docs'
warning: no previously-included files matching '*.pyc' found under directory 'testsuite'
warning: no previously-included files matching '*.pyo' found under directory 'testsuite'
no previously-included directories found matching 'docs/_build'
Got pycodestyle 2.4.0.
Version and requirements information containing pycodestyle:
  Requirement of flake8-print: pycodestyle
  Requirement of flake8-debugger: pycodestyle
  Requirement of flake8==3.5.0: pycodestyle<2.4.0,>=2.0.0
While:
  Installing code-analysis.
Error: There is a version conflict.
We already have: pycodestyle 2.4.0
but flake8 3.5.0 requires 'pycodestyle<2.4.0,>=2.0.0'.
```

Any idea why?

**Comment by sneridagh** [comment]
@tisto tests pass now.

**Pull request merged**